### DISCUSSION TITLE
Documentation and examples

### DISCUSSION THREAD

**Discussion by offwork** [IBM/carbon-components-angular#1805] [Q&A]
Hello, I am Kerem. Thank you very much to the carbon team for their successful work. We decided to start our new project with carbon but, but I have to say that I have had a hard time for the last few days. Especially about the carbon documents because I don't think the storybook alone is enough for documentation.

I spent a lot of time creating a new custom theme however was only able to do this at [the component level](https://carbon-elements.netlify.app/themes/examples/preview/#theming).

Another issue is UIShell. I could not find any examples or information on the usage of [headers](https://github.com/IBM/carbon-components-angular/blob/master/src/ui-shell/sidenav/sidenav.component.ts#L21) and [footers](https://github.com/IBM/carbon-components-angular/blob/master/src/ui-shell/sidenav/sidenav.component.ts#L42) in the Sidenav component.

Finally, is it possible to see examples in places such as [stackbiltz](https://stackblitz.com/) to make these issues clear?

**Comment by mak009** [comment]
I have same query regarding usage of carbon components. Document contains only theories not component code example with validations, error handling etc.

**Comment by cal-smith** [comment]
Hi!

In addition to our storybook we also have some API docs, located here: https://angular.carbondesignsystem.com/documentation/

We definitely don't have as much documentation as we would like. It's a tough balance between fixing bugs and writing docs, at the moment we're still focused on the former so the latter definitely suffers.

We 100% welcome any community documentation contributions, so feel free to open PRs that fill in the gaps!

**Comment by Michael-Baylis** [comment]
What is missing are SIMPLE examples of the various components.   I am trying to write a demo app using Carbon Angular and wanted to add a simple data table.   Looking at the story book:-
```
<ibm-table-container>
			<ibm-table-header>
				<h4 ibmTableHeaderTitle>{{title}}</h4>
				<p ibmTableHeaderDescription>{{description}}</p>
			</ibm-table-header>
			<!--
				app-* components are for demo purposes only.
				You can create your own implementation by using the component source as an example.
			-->
			<app-table
				[model]="model"
				[stickyHeader]="stickyHeader"
				[size]="size"
				[skeleton]="skeleton"
				[enableSingleSelect]="enableSingleSelect"
				[showSelectionColumn]="showSelectionColumn"
				[striped]="striped"
				[sortable]="sortable"
				[isDataGrid]="isDataGrid">
			</app-table>
		</ibm-table-container>
```
So here we have app-table,   which I can understand why you want to do that,  as it encapsulates the data in the demo,   but it is no help as an direct example.
Especially as I can't easily find the source of app-table.
I have been through the documentation sites several times now, and can't find a SIMPLE data table to get me started.
So I have to spend time hunting around for the source of app-table in the hopes that it will give me the simple example

**Reply by Michael-Baylis** [reply]
Not really an answer,  but don't have a easy way to say I agree with the OP

**Reply by armper** [reply]
We're writing our own component showcase with code examples. It's a lot of work :)
